cmd/embedded-cluster/install.go

```diff
@@ -266,10 +266,10 @@ func getLicenseFromFilepath(licenseFile string) (*kotsv1beta1.License, error) {
 		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
 		return nil, fmt.Errorf("license app %s does not match binary app %s, please provide the correct license", license.Spec.AppSlug, rel.AppSlug)
 	}
-	if rel.ChannelID != license.Spec.ChannelID {
+	if !channelIDExists(license, rel.ChannelID) {
 		// if the channel is different, we will not be able to install the pinned vendor application version within kots
 		// this may result in an immediate k8s upgrade after installation, which is undesired
-		return nil, fmt.Errorf("license channel %s (%s) does not match binary channel %s, please provide the correct license", license.Spec.ChannelID, license.Spec.ChannelName, rel.ChannelID)
+		return nil, fmt.Errorf("license channel does not match binary channel %s, please provide the correct license", rel.ChannelID)
```

Comment:
I would still prefer a friendlier error message here, ideally including what channels the license DOES allow, and possibly adding the rel.ChannelSlug as a friendlier thing to look for if that's embedded in the binary (and I believe it is)

Reply:
@laverya 👍 yea you're right. let me send along another update.

Reply:
@laverya ok mind taking another peek ? https://github.com/replicatedhq/embedded-cluster/pull/1220/commits/6467fae350bee3a339d30ae12ccd2b799a81c6a3